### ISSUE TITLE
bl-welcome fails, if installed w/o internet

### ISSUE THREAD

**Issue by Zeitvertreib** [BunsenLabs/bunsen-netinstall#12]
When bl-welcome pops up, the question if contrib and non-free should be enables gives on CR:
"Editing sources list failed. This script will exit.
[...] check the entries before running bl-welcome again."

I then edit the sources to my needs, but thats feels a lot like a bug.

Because i want BL, i installed it from the img (USB), it should provide a source list that works after installation, no matter if installed with or w/o internet.

(I reproduced that circumstance at least twice. Also, big fan, love your work :) )

**Comment by johnraff** [comment]
@Zeitvertreib thank you for this bug report.
In fact, it appears to be two bugs:
1) The iso install fails to provide a sources.list with Debian contrib and non-free repositories enabled. (This was an old bug, which we believed fixed with the current iso)
2) If run with no internet connection, bl-welcome detects the missing entries  but fails to correct them.

Is this correct?

Also, I wonder if you could paste in the contents of bl-welcome's logfile? It's at `$HOME/.cache/bunsen-welcome/bl-welcome.log` Alternatively, if it's too big, maybe you could upload it to a pastebin and post a link, or paste in the section just before it exits?

Thank you for your help.

**Comment by Zeitvertreib** [comment]
*1) the iso was from yesterday, SHA-256 is the same as on your website by the time i write this.
*2) detection: yes, correction: no (fail or missing attempt: unknown) 

This [one](http://pastebin.com/XT7sZMLP) on machine A I didnt bother and did my stuff without rerunning bl-welcome (its the end of the log). 

Edit: No, wait the second one (which i deleted), was just installed w internet. I dont have that log anymore, i experimented w Nvidia drivers, i had to reinstall and formatted the whole thing.. this time w internet. (i hope this helps anyways)

**Comment by johnraff** [comment]
Which specific iso did you install from? I'll try to reproduce the issue.

**Comment by mangstadt** [comment]
I'm having the same problem.  I also installed BunsenLabs without an internet connection (because it couldn't establish a DHCP connection with my Wi-Fi router for some reason).

I used the "bl-Hydrogen-i386.iso" ISO.

Here is the contents of my bl-welcome.log file: https://gist.github.com/mangstadt/a56874939b7bb5dfe3f0243c56fd59b0

Side note: Some of the optional packages have failed as well.  For example, when I try to install **Java Support**, it says:

```
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 bunsen-meta-java : Depends: openjdk-7-jre but it is not going to be installed
                    Depends: icedtea-7-plugin but it is not installable

  There were problems installing bunsen-meta-java

  Press "d" for details (q to quit) d


E: Unable to correct problems, you have held broken packages.
```

Thank you.

**Comment by mangstadt** [comment]
Update: My issue was solved, see: https://forums.bunsenlabs.org/viewtopic.php?pid=28378#p28378

**Comment by johnraff** [comment]
@mangstadt thank you for posting bl-welcome.log

I understand you have solved your problem by editing sources.list manually, but I wonder if you could check for the existence of /etc/apt/sources.list~bl-welcome-_some_number_~ and post its contents here? (It should be quite short.) 

I would like to get some idea why your sources.list was corrupted, and why bl-welcome failed to fix it.
Thanks again for your help.

**Comment by mangstadt** [comment]
@johnraff Sure, no problem.  Thank you for investigating further.

https://gist.github.com/mangstadt/2580df39ebd6333a48d6cb757c90ab11

**Comment by johnraff** [comment]
Thank you. All of these are missing the usual debian entries. I can only assume that this is because the installation was done without an internet connection.

I will try a test installation, but it seems likely that this is the behaviour of the Debian installer which underlies the BunsenLabs installation. Maybe the welcome script can be adjusted to deal with such a situation.

**Comment by mangstadt** [comment]
From what I remember, the Debian installer didn't make it easy to proceed without an internet connection.   Maybe I ended up selecting some specific combination of options that prevented the entries from being added.

Also note that I did NOT use the graphical installer (because my trackpad wouldn't work).

Thank you again for looking into the problem.

**Comment by Zeitvertreib** [comment]
Soo, I installed from bl-Hydrogen-amd64_20160710.iso on Dec.18th.2016.
cat $HOME/.cache/bunsen-welcome/bl-welcome.log
`  -------------------------------[ page 3 of 19 ]--------------------------------

  CHECK REPOSITORIES
  ------------------
  
  Checking apt sources list for possible missing components...
  
  A normal BunsenLabs setup uses the 'contrib' and 'non-free' components of the Debian repositories, but they seem to be 
  missing from your apt sources list. Without them you will be unable to install some packages, for example the Flash Plugins. 
  Would you like to add 'contrib' and 'non-free' to your sources list?
  
enable "contrib" and "non-free"? [Y/n] 
  Editing sources list failed. This script will exit.
  Please open /etc/apt/sources.list in a text editor and check the entries before running bl-welcome again.

Press any key to continue.
`
I post this, because I opened this thread related to the previous iso. And because I installed bl on my older machine yesterday as well, same issue.

I can pretty much life with the current state, copy and paste a sources file after installation dont hurt much. So I don't ask for help now, I just want to point out, thats still there (with the new iso). (And its prob somewhere in the forums as well. And I'm a bit sry, that I abandoned this thread after I was asked something.)

-------------------------
edit: Note: the check repositories is offensive bold, because of the git markdown thing. its just written in caps in the file, and this is what the markdown is makeing of it, its not intentional.. regards

**Comment by darknodi** [comment]
@johnraff the bunsenlabs, based, on linux architecture kernel, framework apt :) 

from installation complete, usb drive live bootable, 

the bootloader don't tells the xfce desktop

**Comment by johnraff** [comment]
@darknodi After installing BunsenLabs on a particular partition, the grub boot screen will not offer any choice between different Desktop Environments that might be installed on that partition. However, the login screen (LightDM by default) will offer a choice between BunsenLabs and other DE's, if any are installed.

**Comment by darknodi** [comment]
what's the differencing have from each others bunsenlabs from crunchbang. 
the openbox desktop, may I have see, an visual graphic interface ( for desktop computers ), 
little, distincly, stable, faster, for workjobs, for workpersonal :) 

I have installed the bunsenlabs
from primary & beggining ext file>
swap logical and beggining. 
space free. 

the startx ( does not start graphical interface ) 
but, I have entered from PPYT1 from login shell 

sudo apt update
sudo apt upgrade 
sudo apt dist-upgrade
sudo apt full-upgrade 

sudo apt install xfce4 

i have opened the interface, 

with this command from shell pppty 

startxfce4

send me the command that to change visual interface from xfce4 to openbox :) 

@johnraff
